AirCasting/ABConnector/SDSyncController.swift

```diff
@@ -57,24 +62,32 @@ class SDSyncController: ObservableObject {
                 case .success(let metadata):
                     progress(.finalizing)
                     let files = self.fileWriter.finishAndSave()
-                    self.checkFilesForCorruption(files, expectedMeasurementsCount: metadata.expectedMeasurementsCount)
-                    //TODO: Continue processing SD card data when files are not corrupted. Return an error and finish sync without clearing sd card if they are.
-                    if let mobileFileURL = files.first(where: { $0.1 == SDCardSessionType.mobile })?.0 {
-                        self.mobileSessionsSaver.saveDataToDb(fileURL: mobileFileURL, deviceID: sensorName) { result in
-                            switch result {
-                            case .success(let sessions):
-                                self.averagingService.averageMeasurements(for: sessions) {
-                                    Log.info("Averaging done")
-                                    self.onCurrentSyncEnd { self.startBackendSync() }
+                    
+                    //MARK: checking if files have the right number of rows and if rows have the right values
+                    self.checkFilesForCorruption(files, expectedMeasurementsCount: metadata.expectedMeasurementsCount) { fileValidationResult in
```

Comment:
this part got huuuuge. After we release and can do some refactoring I'd like this class to use `Combine` in order to make it nice and flat